graylog2-web-interface/src/components/search/MessageTableEntry.jsx

```diff
@@ -102,6 +113,12 @@ const MessageTableEntry = React.createClass({
   _toggleDetail() {
     this.props.toggleDetail(`${this.props.message.index}-${this.props.message.id}`);
   },
+
+  _toUserTime(value) {
+    const dateTime = new DateTime(value);
+    return dateTime.toString(DateTime.Formats.TIMESTAMP);
```

Comment:
Maybe it makes sense to wrap the `value` in a `Timestamp` component instead and pass the timezone of the user as a prop? This way we have a more central handling of the time conversion.

Reply:
After the changes the `timestamp` field in messages is in the user defined time zone (or the one set in the browser), which is what the user expects. Unfortunately, other timestamps may still come in UTC, e.g. if there's one in the `full_message` field. Additionally, when adding the `timestamp` field to the search query, the value we set there is using UTC (because we already format the time to be usable against ES), and I find that a bit confusing. Here's an screenshot to illustrate the problem:

![image 2018-01-26 13-44-46](https://user-images.githubusercontent.com/716185/35440343-44d5af6e-029f-11e8-9635-11d9a0c8d11b.png)

To reduce this confusion I would use `DateTime.Formats.TIMESTAMP_TZ` to format the `timestamp` field, making the user aware that this date time is displayed in a certain time zone, and others may not. We could also add a little informative text somewhere next to the value, but only if we find a way while not making it too prominent.

Reply:
Yes, I agree. This would also mean that for consistency we should format the timestamp column in the message table (!= message details) accordingly.

Reply:
> Maybe it makes sense to wrap the value in a Timestamp component instead and pass the timezone of the user as a prop?
I do not know how to archive this without breaking `possiblyHighlight` which needs a String to operate.

> We could also add a little informative text somewhere next to the value
I would like that. At first I thought about a mouseover popup, but you would need to know that, to get the information.
Then I thought about to add the text somewhere on the right side in bright gray but that seems to distract more then I thought :(